cnxepub/templates/exercise_template.xhtml.jinja

```diff
@@ -0,0 +1,56 @@
+<div 
+    data-type="injected-exercise"
+    data-injected-from-nickname="{{ data.nickname }}"
+    data-injected-from-version="{{ data.version }}"
+    data-injected-from-url="{{ data.url }}"
+    data-tags="{{ ' '.join(data.tags) }}"
+    data-is-vocab="{{ data.is_vocab }}"
+>
+    {% if data.required_context %}
+    <div data-type="required-context"
+        data-context-module="{{ data.required_context.module }}"
+        data-context-feature="{{ data.required_context.feature }}">
+        <a class="autogenerated-content"
+        href="#{{ data.required_context.ref }}">[link]</a>
+    </div>
+    {% endif %}
+    {% if data.stimulus_html %}
+    <div data-type="exercise-stimulus">{{ data.stimulus_html }}</div>
+    {% endif %}
+    {% for question in data.questions %}
+    <div 
+        data-type="question"
+        data-is-answer-order-important="{{ question.is_answer_order_important }}"
+        data-formats="{{ ' '.join(question.formats) }}"
+    >
+        {% if question.stimulus_html %}
+        <div data-type="question-stimulus">{{ question.stimulus_html }}</div>
+        {% endif %}
+        <div data-type="question-stem">{{ question.stem_html }}</div>
+        {% if question.answers %}
+        <ol data-type="multiple-choice-options" type="a">
+            {% for option in question.answers %}
+            <li
+                data-type="option"
+                data-id="{{ option.id }}"
+                data-correctness="{{ option.correctness }}"
+            >
+                <div data-type="option-content">{{ option.content_html }}</div>
+                {% if option.feedback_html %}<div data-type="option-feedback">{{ option.feedback_html }}</div>{% endif %}
+            </li>
+            {% endfor %}
+        </ol>
+        {% endif %}
+        {% for solution in question.collaborator_solutions %}
+        <div data-type="solution" data-solution-source="collaborator" data-solution-type="{{ solution.solution_type }}">
+            {{ solution.content_html }}
+        </div>
+        {% endfor %}
+        {% for solution in question.community_solutions %}
+        <div data-type="solution" data-solution-source="community" data-solution-type="{{ solution.solution_type }}">
+            {{ solution.content_html }}
+        </div>
+        {% endfor %}
```

Comment:
@phil you mentioned putting the solutions inside a list, but I wonder if it may be uncommon for these questions to each have more than one solution; if that's the case, it might be better to leave them as-is and let the recipe handle formatting multiple solutions